internal/pkg/crawl/sitespecific/youtube/youtube.go

```diff
@@ -0,0 +1,35 @@
+package youtube
+
+import (
+	"io"
+	"net/url"
+
+	"github.com/internetarchive/Zeno/internal/pkg/crawl/dependencies/ytdlp"
+)
+
+func Parse(body io.ReadCloser) (URLs []*url.URL, err error) {
+	// Create a temporary server to serve the body and call ytdlp on it
+	port, stopChan, err := ytdlp.ServeBody(body)
+	if err != nil {
+		return nil, err
+	}
+	defer close(stopChan)
```

Comment:
maybe close the channel in `internal/pkg/crawl/dependencies/ytdlp/server.go#L32` and just send a `struct{}{}` here, it feels more logical to me